components/ee/agent-smith/pkg/agent/agent.go

```diff
@@ -211,12 +213,29 @@ func (agent *Smith) Start(ctx context.Context, callback func(InfringingWorkspace
 		clo = make(chan classifiedProcess, 50)
 	)
 	agent.metrics.RegisterClassificationQueues(cli, clo)
+
+	workspaces := make(map[int]*common.Workspace)
+	wssWriteMutex := &sync.Mutex{}
+
 	defer wg.Wait()
 	for i := 0; i < 25; i++ {
 		wg.Add(1)
 		go func() {
 			defer wg.Done()
 			for i := range cli {
+				// Update the workspaces map if this process belongs to a new workspace
```

Comment:
I have clubbed both classification and workspaces map creation in this go routine.

I cannot think of a better option. Other option was to run a separate process detector and consume its out put in a dedicated go routine. process detectors are expensive so I refrained from doing that.